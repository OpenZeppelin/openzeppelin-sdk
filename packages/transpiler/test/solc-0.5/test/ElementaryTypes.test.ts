import { shouldTranspileToValidContract } from './LocalConract.behaviour';

shouldTranspileToValidContract('ElementaryTypes');
