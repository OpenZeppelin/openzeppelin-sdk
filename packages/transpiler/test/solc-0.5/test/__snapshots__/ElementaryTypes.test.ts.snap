// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementaryTypes contract is converted to a valid ElementaryTypesUpgradable contract 1`] = `
"pragma solidity ^0.5.0;
import \\"@openzeppelin/upgrades/contracts/Initializable.sol\\";

contract ElementaryTypesUpgradable is Initializable  {
    function initialize() external initializer {
        __init(true);
    }

    function __init(bool callChain) internal {
        
        
        owner = address(0x123);
        active = true;
        hello = \\"hello\\";
        count = -123;
        ucount = 123;
        samevar = \\"stringliteral\\";
        
    }

    address public owner ;
    bool active ;
    string hello ;
    int count ;
    uint ucount ;
    bytes32 samevar ;
}
"
`;
